### PR TITLE
feat(route-href): add custom attribute for generating hrefs for named routes

### PR DESCRIPTION
**Pull request by bryanrsmith**
fixes aurelia/router#4

Depends on aurelia/router#67

Feedback welcome!

### CODE REVIEW

File: src/route-href.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
+import {customAttribute, bindable} from 'aurelia-templating';
+import {inject} from 'aurelia-dependency-injection';
+import {Router} from 'aurelia-router';
+import {TaskQueue} from 'aurelia-task-queue';
```
Comment (bryanrsmith) [inline src/route-href.js]: This is just to coalesce changes from the 3 bindable change handlers that fire with initial values.

Are the 'Dependencies' and 'Used by' README sections manually maintained?  If so we'll need to update this repo, and aurelia/task-queue.

File: src/route-href.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    if (!this.isChangePending) {
+      this.isChangePending = true;
+      this.taskQueue.queueMicroTask(this.updateAttribute.bind(this));
+    }
+  }
+
+  updateAttribute() {
+    this.isChangePending = false;
+
+    let href = this.router.generate(this.route, this.params);
```
Comment (bryanrsmith) [inline src/route-href.js]: What kind of error handling is desirable here? This call could throw if the router doesn't recognize the route, or the params don't satisfy the route's dynamic segment requirements.

Comment (EisenbergEffect) [inline src/route-href.js]: What was the reason for queuing the micro task?

File: src/route-href.js (hunk 1 above)
Comment (EisenbergEffect) [inline src/route-href.js]: Those sections are maintained manually at the moment, yes.

File: src/route-href.js (hunk 2 above)
Comment (bryanrsmith) [inline src/route-href.js]: Noted above,

> This is just to coalesce changes from the 3 bindable change handlers that fire with initial values.

The 3 change handlers all fire with initial values, and I only wanted to do the work once when all values had been set. Is there a simpler way to accomplish this?

Comment (EisenbergEffect) [inline src/route-href.js]: Yes. You can implement the `bind` callback. When you have this, your property change callbacks will not be fired individually during the initial bind, but the `bind` callback will be called once everything is set up. Afterwords, the individual callbacks will be fired as the individual properties change. Does that make sense?

Comment (bryanrsmith) [inline src/route-href.js]: Ah, thank you. I didn't realize that implementing `bind` would prevent the initial calls to change handlers. I'll make the change this evening.

**Pull request merged**